clash-lib/src/Data/Aeson/Extra.hs

```diff
@@ -55,8 +58,13 @@ decodeOrErr
   -> a
 decodeOrErr path contents =
   case parse json contents of
-    Done _ v ->
+    Done leftover v ->
       case fromJSON v of
+        Success a | BS.any notWhitespace leftover ->
+          trace ("WARNING: After a successfully parsing: " ++ show path
+                 ++ " there was some trailing garbage leftover, ignoring:\n"
+                 ++ BSChar.unpack leftover)
+                a
```

Comment:
Although an improvement over the current situation, this should be a hard error IMO.